### PR TITLE
simplify ci_is_equal

### PR DESCRIPTION
**Pull request by alandefreitas**

### CODE REVIEW

**Comment by alandefreitas** [comment]
This function was duplicated.

I was looking for bad pointer operations #467 but I couldn't find any. 

But I ended up simplifying this while doing it.

File: include/boost/url/grammar/detail/impl/ci_string.ipp [hunk 1]
```diff
@@ -22,36 +22,7 @@ namespace detail {
...
-        if( to_lower(a) !=
-            to_lower(b))
-            return false;
-        a = *p1++;
-        b = *p2++;
-    }
-    while(n--);
-    return true;
-}
+extern template bool ci_is_equal<string_view, string_view>(string_view const&, string_view const&);
```
Comment (vinniefalco) [inline include/boost/url/grammar/detail/impl/ci_string.ipp:25]: Why is this line so long

File: include/boost/url/grammar/detail/ci_string.hpp [hunk 2]
```diff
@@ -140,32 +134,34 @@ ci_is_equal(
...
         detail::type_id<S1>());
 
-    auto it0 = s0.begin();
-    auto it1 = s1.begin();
-    auto const end0 = s0.end();
-    auto const end1 = s1.end();
-    for(;;)
+    auto n = s0.size();
+    auto p1 = s0.begin();
+    auto p2 = s1.begin();
```
Comment (vinniefalco) [inline include/boost/url/grammar/detail/ci_string.hpp:139]: p0 and p1... why did you renumber them

File: include/boost/url/grammar/detail/ci_string.hpp [hunk 3]
```diff
@@ -140,32 +134,34 @@ ci_is_equal(
...
+    char a, b;
```
Comment (vinniefalco) [inline include/boost/url/grammar/detail/ci_string.hpp:140]: c0, c1? heh

**Comment by vinniefalco** [comment]
Was this merged?

**Comment by alandefreitas** [comment]
> Was this merged?

No. You told me to give up on this.

**Comment by vinniefalco** [comment]
The change to `all_chars_t` is good, we should keep that

**Comment by vinniefalco** [comment]
Was this the thing with the problem of not having `size()` on a range?

**Comment by alandefreitas** [comment]
> Was this the thing with the problem of not having size() on a range?

Yes. This is related to the problem here. And that's why you told me to keep both functions.

**Pull request closed without merging**